.github/workflows/unit-test-on-pull-request.yml

```diff
@@ -38,7 +38,7 @@ jobs:
         id: linter-version
         run: (echo -n "version="; make linter-version) >> "$GITHUB_OUTPUT"
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@55c2c1448f86e01eaae002a5a3a9624417608d84 # v6
+        uses: golangci/golangci-lint-action@4afd733a84b1f43292c63897423277bb7f4313a9 # v8.0.0
```

Comment:
This conflicts with our Makefile:

https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/2a2d56bae8f62092bfe194b6524170acd067d265/Makefile#L86

Reply:
Automatically updating golangci-lint is possibly not a good idea, as new version may introduce new linters with positives, which require manual fixes.
I'd suggest to use `make lint` in the CI and update the linter version in the `Makefile` every now and then (e.g. with every new minor Go version).

Reply:
That's OK as a quick fix.
Maybe in a separate PR: Add a CI step calling `make lint`